shared/src/Datadog.Trace.ClrProfiler.Native/crashhandler.cpp

```diff
@@ -0,0 +1,409 @@
+        
+        if (result == ERROR_SUCCESS)
+        {
+            DWORD dataSize = 0;
+            DWORD valueType = 0;
+            result = RegQueryValueEx(hKey, valueName, NULL, &valueType, NULL, &dataSize);
+            RegCloseKey(hKey);
+
+            return result == ERROR_SUCCESS;
+        }
+
+        return false;
+    }
+
+    // Capture all the environment variables starting with "DD_"
+    void GetDatadogEnvironmentBlock(WCHAR*& environmentVariables, int32_t& length)
+    {
+        auto envStrings = GetEnvironmentStrings();
+
+        std::wstring envBlock;
+
+        if (envStrings == nullptr)
+        {
+            length = 2;
+            environmentVariables = new WCHAR[length];
+            environmentVariables[0] = L'\0';
+            environmentVariables[1] = L'\0';
+            return;
+        }
+
+        for (LPWCH env = envStrings; *env != L'\0'; env += wcslen(env) + 1)
+        {
+            if (wcsncmp(env, L"DD_", 3) == 0)
+            {
+                envBlock.append(env);
+                envBlock.push_back('\0');
+            }
+        }
+
+        // Environment block ends with a double null terminator
+        envBlock.push_back('\0');
+
+        if (envBlock.length() == 1)
+        {
+            // If the environment block was empty, we're still missing one null terminator
+            envBlock.push_back('\0');
+        }
+
+        length = static_cast<int32_t>(envBlock.length());
+        environmentVariables = new WCHAR[length];
+        memcpy(environmentVariables, envBlock.c_str(), length * sizeof(WCHAR));
+        FreeEnvironmentStrings(envStrings);
+    }
+
+    // Merge two environment blocks. The first one takes precedence.
+    std::vector<WCHAR> MergeEnvironmentBlocks(LPCWSTR envBlock1, LPCWSTR envBlock2)
+    {
+        // Define a case-insensitive comparator for keys
+        struct CaseInsensitiveCompare
+        {
+            bool operator()(const std::wstring& lhs, const std::wstring& rhs) const
+            {
+                return _wcsicmp(lhs.c_str(), rhs.c_str()) < 0;
+            }
+        };
+
+        // Map to hold environment variables with case-insensitive keys
+        std::map<std::wstring, std::wstring, CaseInsensitiveCompare> envMap;
+
+        // Helper lambda to parse an environment block into the map
+        auto parseEnvBlock = [&envMap](LPCWSTR envBlock)
+            {
+                if (envBlock)
+                {
+                    LPCWSTR curr = envBlock;
+                    while (*curr)
+                    {
+                        std::wstring entry = curr;
+                        size_t pos = entry.find(L'=');
+                        if (pos != std::wstring::npos)
+                        {
+                            std::wstring key = entry.substr(0, pos);
+                            std::wstring value = entry.substr(pos + 1);
+                            envMap[key] = value; // Insert or overwrite the key
```

Comment:
nit: you can move the key and value

Reply:
I'm adding move for the value. It feels weird for the key IMO because we don't store it